spdx/v2/v2_2/package.go

```diff
@@ -113,8 +115,37 @@ type Package struct {
 	Files []*File `json:"files,omitempty"`
 
 	Annotations []Annotation `json:"annotations,omitempty"`
+
+	// this field is only used when decoding JSON to translate the hasFiles
+	// property to relationships
+	hasFiles []common.DocElementID
+}
+
+func (p *Package) UnmarshalJSON(b []byte) error {
+	type pkg Package
+	type extras struct {
+		HasFiles []common.DocElementID `json:"hasFiles"`
```

Comment:
ahhh i see this trick to pass the `hasFiles` up to unexposed fields, this is very cool :)